src/components/button-swap/button-swap.test.ts

```diff
@@ -0,0 +1,177 @@
+import path from 'path';
+import type { PleasantestUtils } from 'pleasantest';
+import { withBrowser } from 'pleasantest';
+import { loadTwigTemplate, loadGlobalCSS } from '../../../test-utils';
+
+// Helper to load the Twig template file
+const componentMarkup = loadTwigTemplate(
+  path.join(__dirname, './button-swap.twig')
+);
+
+// Helper to initialize the component JS
+const initJS = (
+  utils: PleasantestUtils,
+  initialCallback?: () => void,
+  swappedCallback?: () => void
+) =>
+  utils.runJS(
+    `
+    import { initButtonSwap } from './button-swap'
+    export default (initialCallback, swappedCallback) => {
+      initButtonSwap(document.querySelector('.js-button-swap'), {
+        initialCallback,
+        swappedCallback
+      })
+    }
+    `,
+    [initialCallback, swappedCallback]
+  );
+
+test(
+  'Initial state',
+  withBrowser(async ({ utils, screen }) => {
+    await utils.injectHTML(await componentMarkup());
+    await loadGlobalCSS(utils);
+
+    // Visually hidden text for a more inclusive UX
+    const statusMsg = await screen.getByRole('status');
+    await expect(statusMsg).toHaveTextContent(
+      /^currently unsubscribed from notifications$/i
+    );
+
+    const statusMsgs = await screen.getAllByRole('status');
+    expect(statusMsgs.length).toBe(1);
```

Comment:
You can use just `getByRole` since `getBy` matchers will throw if there are 0 results or more than one result, it will only pass if there is exactly 1: https://testing-library.com/docs/dom-testing-library/cheatsheet#queries

Reply:
Thank you! 🎓

Reply:
Done! https://github.com/cloudfour/cloudfour.com-patterns/pull/1516/commits/2ddbb739661f82016db66aad11fe6d30b269d2bc